mdn/src/test/java/org/apache/james/mdn/MDNTest.java

```diff
@@ -215,6 +229,196 @@ void mime4JMessageExportShouldGenerateExpectedContentType() throws Exception {
             .containsPattern(Pattern.compile("Content-Type: multipart/report;.*(\r\n.+)*report-type=disposition-notification.*(\r\n.+)*\r\n\r\n"));
     }
 
+    @Test
+    public void parseShouldThrowWhenNonMultipartMessage() throws Exception {
+        Message message = Message.Builder.of()
+            .setBody("content", StandardCharsets.UTF_8)
+            .build();
+        assertThatThrownBy(() -> MDN.parse(message))
+            .isInstanceOf(MDN.MDNParseContentTypeException.class)
+            .hasMessage("MDN Message must be multipart");
+    }
+
+    @Test
+    public void parseShouldThrowWhenMultipartWithSinglePart() throws Exception {
+        Message message = Message.Builder.of()
+            .setBody(MultipartBuilder.create()
+                .setSubType("report")
+                .addTextPart("content", StandardCharsets.UTF_8)
+                .build())
+            .build();
+        assertThatThrownBy(() -> MDN.parse(message))
+            .isInstanceOf(MDN.MDNParseBodyPartInvalidException.class)
+            .hasMessage("MDN Message must contain at least two parts");
+    }
+
+    @Test
+    public void parseShouldThrowWhenSecondPartWithBadContentType() throws Exception {
+        Message message = Message.Builder.of()
+            .setBody(MultipartBuilder.create()
+                .setSubType("report")
+                .addTextPart("first", StandardCharsets.UTF_8)
+                .addTextPart("second", StandardCharsets.UTF_8)
+                .build())
+            .build();
+        assertThatThrownBy(() -> MDN.parse(message))
+            .isInstanceOf(MDN.MDNParseException.class)
+            .hasMessage("MDN can not extract");
+    }
+
+    @Test
+    public void parseShouldFailWhenMDNMissingMustBeProperties() throws Exception {
+        Message message = Message.Builder.of()
+            .setBody(MultipartBuilder.create("report")
+                .addTextPart("first", StandardCharsets.UTF_8)
+                .addBodyPart(BodyPartBuilder
+                    .create()
+                    .setBody(SingleBodyBuilder.create()
+                        .setText("Final-Recipient: rfc822; final_recipient\r\n" +
+                            "".replace(System.lineSeparator(), "\r\n").strip())
```

Comment:
That looks complicated. Is `"".replace(System.lineSeparator(), "\r\n").strip()` needed?